### PR TITLE
Add Hincrby command

### PR DESCRIPTION
**Pull request by icemint0828**
Related to #157 

* Add tests.
* Add a server command for hincrby.

Please review when you have time!

### CODE REVIEW

File: hash.go [hunk 1]
```diff
@@ -314,3 +319,53 @@ func (db *RoseDB) HScan(key []byte, prefix []byte, pattern string, count int) ([
 	}
 	return values, nil
 }
+
+// HIncrby increments the number stored at field in the hash stored at key by increment.
+// If key does not exist, a new key holding a hash is created.
+// If field does not exist the value is set to 0 before the operation is performed.
+// The range of values supported by HINCRBY is limited to 64bit signed integers.
```
Comment (gozeloglu) [inline hash.go:326]: You can move a partial upper line at the beginning of this line to use the space efficiently.

```suggestion
// HIncrby increments the number stored at field in the hash stored at key by increment.
// If key does not exist, a new key holding a hash is created. If field does not exist
// the value is set to 0 before the operation is performed. The range of values supported
//  by HINCRBY is limited to 64bit signed integers.
```

Comment (icemint0828) [reply hash.go:326]: @gozeloglu 
Thank you for your suggestion.
It's ok!

File: cmd/command.go [hunk 2]
```diff
@@ -505,6 +505,18 @@ func hScan(cli *Client, args [][]byte) (interface{}, error) {
 	return cli.db.HScan(args[0], args[1], pattern, count)
 }
 
+func hIncrby(cli *Client, args [][]byte) (interface{}, error) {
```
Comment (gozeloglu) [inline cmd/command.go:508]: It is good to follow Go naming convention. 
```suggestion
func hIncrBy(cli *Client, args [][]byte) (interface{}, error) {
```

File: hash.go [hunk 3]
```diff
@@ -314,3 +319,53 @@ func (db *RoseDB) HScan(key []byte, prefix []byte, pattern string, count int) ([
 	}
 	return values, nil
 }
+
+// HIncrby increments the number stored at field in the hash stored at key by increment.
+// If key does not exist, a new key holding a hash is created. If field does not exist
+// the value is set to 0 before the operation is performed. The range of values supported
+//  by HINCRBY is limited to 64bit signed integers.
+func (db *RoseDB) HIncrby(key, field []byte, incr int64) (int64, error) {
```
Comment (gozeloglu) [inline hash.go:327]: It is good to follow Go naming convention.

```suggestion
func (db *RoseDB) HIncrBy(key, field []byte, incr int64) (int64, error) {
```

File: hash.go [hunk 4]
```diff
@@ -314,3 +319,53 @@ func (db *RoseDB) HScan(key []byte, prefix []byte, pattern string, count int) ([
 	}
 	return values, nil
 }
+
+// HIncrby increments the number stored at field in the hash stored at key by increment.
```
Comment (gozeloglu) [inline hash.go:323]:
```suggestion
// HIncrBy increments the number stored at field in the hash stored at key by increment.
```

File: hash_test.go [hunk 5]
```diff
@@ -647,3 +648,79 @@ func TestRoseDB_HScan(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, 8, len(values))
 }
+
+func TestRoseDB_HIncrby(t *testing.T) {
```
Comment (gozeloglu) [inline hash_test.go:652]:
```suggestion
func TestRoseDB_HIncrBy(t *testing.T) {
```

**Review by gozeloglu — COMMENTED** [review]
Sorry for missing these small details lately.

**Review by gozeloglu — APPROVED** [review]
```
   __   _____________  ___
  / /  / ___/_  __/  |/  /
 / /__/ (_ / / / / /|_/ / 
/____/\___/ /_/ /_/  /_/  
```

**Comment by icemint0828** [comment]
> Sorry for missing these small details lately.

Don't worry about it!
Sorry I wasn't paying attention for the naming convention of HIncrBy.

There were others, so I fixed them.
Thank you for your review:smile:

File: hash.go [hunk 6]
```diff
@@ -314,3 +319,53 @@ func (db *RoseDB) HScan(key []byte, prefix []byte, pattern string, count int) ([
 	}
 	return values, nil
 }
+
+// HIncrBy increments the number stored at field in the hash stored at key by increment.
+// If key does not exist, a new key holding a hash is created. If field does not exist
+// the value is set to 0 before the operation is performed. The range of values supported
+//  by HINCRBY is limited to 64bit signed integers.
```
Comment (roseduan) [inline hash.go:326]: //    by
A blank before by, which is not necessary

**Pull request merged**